.. _packages:


=====
Our Packages
=====

We have developed the following open-source software packages to analyze high-throughput genomics data with biophysical models. Click on the cards to explore our tools! 

.. grid:: 1 2 3 4

    .. grid-item-card:: *Monod*
       :link: monod
       :link-type: ref
       :text-align: center
   
    .. grid-item-card:: *biVI*
       :link: bivi
       :link-type: ref
       :text-align: center
   
    .. grid-item-card:: meK-Means
       :link: mekmeans
       :link-type: ref
       :text-align: center
   
    .. grid-item-card:: Chronocell
       :link: chronocell
       :link-type: ref
       :text-align: center


.. toctree::
   :hidden:

   packages_monod.rst
   packages_bivi.rst
   packages_mekmeans.rst
   packages_chronocell.rst
   
=========================================

**Generating unspliced and spliced count matrices with kallisto**

Our tools require as inputs spliced and unspliced UMI counts per gene and per cell (two cell by gene matrices). The current version of our alignment software `kallisto <https://pachterlab.github.io/kallisto/>`_ outputs these count matrices, as described in detail in `the protocol preprint <https://www.biorxiv.org/content/10.1101/2023.11.21.568164v2>`_.
 

=========================================
